app/portainer/settings/authentication/internal-auth/PasswordLengthSlider.tsx

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from 'react';
+import RcSlider from 'rc-slider';
+import clsx from 'clsx';
+
+import { Badge } from '@/portainer/components/Badge/Badge';
+
+import 'rc-slider/assets/index.css';
+
+import styles from './PasswordLengthSlider.module.css';
+
+export interface Props {
+  min: number;
+  max: number;
+  step: number;
+  defaultValue: number;
+}
+
+export function PasswordLengthSlider({ min, max, step, defaultValue }: Props) {
+  const SliderWithTooltip = RcSlider.createSliderWithTooltip(RcSlider);
+  const [labelProperties, setLabelProperties] = useState({
+    sliderValue: 0,
+    icon: '',
+    strength: '',
+    labelColor: '',
+  });
+
+  const icons: Record<string, string> = {
+    Weak: 'fa-times-circle',
+    Good: 'fa-check-circle',
+    Strong: 'fa-check-circle',
+    'Very strong': 'fa-lock',
+  };
+
+  const colors: Record<string, string> = {
```

Comment:
I think you should have an enum or a union `type Strength = "weak"|"good"|"strong"|"veryStrong"` and a setting object `Record<Strength, {icon:string, color: string}>`

Reply:
this is very interesting! hopefully i have done it right 👍🏻